docs/blockdata.md

```diff
@@ -3,4 +3,174 @@ id: blockdata
 title: Blockchain Data
 ---
 
-TODO
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+## Introduction
+
+In this guide, we'll explore how to provide chain data to LDK upon startup and
+as new blocks are mined. This allows LDK to maintain channel state and monitor
+for on-chain channel activity.
+
+## Overview
+
+LDK maintains channels with your node's peers during the course of node
+operation. When a new channel is opened, the `ChannelManager` will keep track of
+the channel's state and tell the `ChainMonitor` that a new channel should be
+watched. The `ChainMonitor` does so by maintaining a `ChannelMonitor` for each
+channel.
+
+When a new block is mined, it is connected to the chain while other blocks may
+be disconnected. LDK will process such events as they are fed into it from a
+`BlockSource` by:
+
+* Updating channel state
+* Signaling back transactions to filter
+* Broadcasting transactions if necessary
+
+We will walk through this process as depicted here:
+
+![LDK block processing](assets/ldk-block-processing.svg)
+
+## Block Source
+
+Initially, our node doesn't have any channels and so has no data to monitor for
+on-chain. When a channel is opened with a peer, the `ChannelManager` creates a
+`ChannelMonitor` and passes it to the `ChainMonitor` to watch.
+
+At this point, LDK needs to be fed chain data of interest so that it can respond
+accordingly. It supports receiving either full blocks or pre-filtered blocks.
+Block data can sourced from anywhere, but it is your responsibility to ensure
+that the necessary `block_connected` and `block_disconnected` methods are called
+on `ChannelManager` and `ChainMonitor`.
```

Comment:
Should there be an explanation in parentheses why it needs to be called on both? I think this might confuse people and cause questions such as in what sequence.

Reply:
Added a short statement about these.

@TheBlueMatt I can't recall if the ordering matters anymore. Our test fail if I reorder them, but that may just be brittleness.